teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -59,26 +63,41 @@ There are a couple issues with this:
 
-SPIRE can be used to request certificates (SVIDs) for a given k8s workload and can use these certificates to create signatures for the data we need to verify, i.e. an entire TaskRun yaml and results. For our use case, both the Tekton entrypointer image (running in Pods) and the Tekton Controller will interact with SPIRE.
+Pods running in a cluster can interact with SPIRE through SPIFFE api via the local SPIRE agent socket.
+
+Getting pod identity
+- A pod can request for an identity by talking to the local agent socket. The SPIRE agent attests the identity of the pod,
+and request for an identity for the pod to the SPIRE server.
+- The workload then receives one or more SPIFFE Verifiable Identity Document, or [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid), an X509 certificate together with the associated private key for the pod's identity. The x509 certificate is signed by the SPIRE server's certificate authority.
+
+Verifying pod identity
+- A pod can request for an trust bundle of a SPIRE server, which will include the public key to validate the pod's certificate.
+- This trust bundle can be used to verify the SVID of a pod
+
+Note: Alternatively, using the JWT SPIFFE endpoint, one can request a JWT with a specified audience field. Although possible, it may not be idiomatic to use the audience for payload signing purposes. However, there is an open [issue](https://github.com/spiffe/spire/issues/1848) discussing adding arbitrary claims as well.
+
+Identity registration is an important aspect of getting SPIRE identities. This defines the subject name of identities that are minted to pods, as well as the attestation requirements of the workloads. This can be done specific to usecase, or there is a kubernetes workload registrar that creates identity based on the fully qualified canonical pod name.
+
+### Using SPIRE for provenance
+
+SPIRE can be used for provenance using the private key provided from the workload's SVID and signing a payload. The signature will be verifiable by the x509 certificate of the SVID together with the trust bundle of the SPIRE deployment. The verification would ensure that the payload's provenance comes from the Tekton entrypointer image (running in Pods) or the Tekton Controller.
 
 ### Installing SPIRE
-SPIRE runs as a Unix domain socket on the k8s node.
-We can use [spiffe-csi](https://github.com/spiffe/spiffe-csi) to mount the SPIRE socket into Pods as a `csi` type Volume so that we don't have to rely on the `hostPath` volume.
+- SPIRE server runs as a deployment in kubernetes (for simplicity, we'll assume a single cluster SPIRE deployment).
+- SPIRE agents run as a daemonset in the kubernetes cluster, listening on a Unix domain socket on each k8s node.
+- SPIRE kubernetes workload registrar would be optionally installed to provide automatic registration of SPIRE identities. This is optional and could be handled by the Tekton controller as well.
```

Comment:
could you explain a bit more about what the registration of SPIRE identities means? (or link to a doc)

Reply:
Yup - i'll add this link https://spiffe.io/docs/latest/deploying/registering/.